src/async/timing.ts

```diff
@@ -4,13 +4,13 @@ import Promise from 'dojo-shim/Promise';
  * Used for delaying a Promise chain for a specific number of milliseconds.
  *
  * @param milliseconds the number of milliseconds to delay
- * @return {function(T): Promise<T>} a function producing a promise that eventually returns the value passed to it; usable with Thenable.then()
+ * @return {function(T): Promise<T>} a function producing a promise that eventually returns the value or execute the value function passed to it; usable with Thenable.then()
  */
 export function delay<T>(milliseconds: number): Identity<T> {
 	return function (value: T): Promise<T> {
```

Comment:
Actually, we should type this better, as it doesn't represent what will be returned properly.

It should be `return function (value: T | () => T | Thenable<T>): Promise<T>` which also likely means that I bet `Identity` also needs a change.

I would check the intellisense to validate that what you expect the `Promise<T>` to be is what it actually is, because I think you might have an issue.  Also, I think we should "unwrap" thenables too.